lib/fs.js

```diff
@@ -1579,6 +1579,17 @@ function watch(filename, options, listener) {
   if (listener) {
     watcher.addListener('change', listener);
   }
+  if (options.signal) {
+    if (options.signal.aborted) {
+      process.nextTick(() => watcher.close());
+    } else {
+      const listener = () => watcher.close();
+      options.signal.addEventListener('abort', listener);
```

Comment:
Not sure if that matters, but at least it makes it clearer that this is just a one timer:
```suggestion
      options.signal.addEventListener('abort', listener, { once: true });
```

Reply:
This is a common issue that gets brought up from time to time. The reason I like doing `{ once: true }` less is that it creates the false impression we are relying on the listener being a one-time listener for cleanup.

That assumption has lead to memory leaks in our code before - this way the cleanup is explicit (when the watcher is closed) and I _think_ it makes bugs slightly less likely.